src/SurfSplitRouter.js

```diff
@@ -268,6 +268,29 @@ export const SurfSplitRouter: RouterFactory<
                 if (newState == null) {
                     return null;
                 }
+
+                // Ensure the history always includes the initial route
+                const initialRoute = state.routes.find(
+                    ({ name }) => name === initialRouteName,
+                );
+
+                if (initialRoute) {
+                    // Check if the history contains initial route already
+                    if (
+                        !newState.history ||
```

Comment:
@savelichalex do you think we need to add the initial route into an empty history?
I honestly don't clearly understand in what case the history should be undefined...

Reply:
That's a good question actually, maybe it's just due to uncertainty of either we have history property in current state or not, when in reality it always should be...